### PR TITLE
Bringing bundle up to new Homebrew Rubocop standard

### PR DESCRIPTION
**Pull request by colindean**
- [X] Rubocop violations that can be automatically fixed
- [ ] 72  `RSpec/MessageSpies`
- [ ] 60  `RSpec/ContextWording`
- [ ] 55  `RSpec/MultipleExpectations`
- [ ] 54  `RSpec/AnyInstance`
- [ ] 33  `RSpec/NamedSubject`
- [ ] 23  `RSpec/ExampleLength`
- [ ] 13  `RSpec/ExpectInHook`
- [x] 17  `RSpec/FilePath`
- [ ]  7  `RSpec/NestedGroups`
- [x]  1  `RSpec/SubjectStub`
- [x]  1  `RSpec/VerifiedDoubles`

### CODE REVIEW

**Comment by colindean** [comment]
@MikeMcQuaid 

1. Is this even worth pursuing? 
    1. Should we make a TODO config and chip away at this debt?
    1. Should we identify the recently added cops in the upstream config and disable them?

**Comment by MikeMcQuaid** [comment]
> 1. Is this even worth pursuing?

Yes.

> 1. Should we make a TODO config and chip away at this debt?

Seems like a good idea for those that can't be fixed automatically with `brew style --fix` (which could be fixed immediately).

> 2. Should we identify the recently added cops in the upstream config and disable them?

No, let's stick to the TODO.

**Comment by MikeMcQuaid** [comment]
> * [x]  Rubocop violations that can be automatically fixed

Note that this does not seem to have been bulletproof unfortunately so there are failing tests in the Travis output. My suggestion: fix those up so tests are green and add a `.rubocop_todo.yml` for now and merge it.

**Comment by colindean** [comment]
What's included in this PR's commits so far was _just_ the automatic fixes. The other listed cops are ones that couldn't be automatically fixed and need to be added to the todo.

I'll invest a little time fixing the cops with few violations and TODO the remainder.

ETA 10/24 evening EDT unless something else comes up

**Comment by MikeMcQuaid** [comment]
@colindean I've temporarily disabled `brew style` and moved us to Azure Pipelines for CI. Can you uncomment these lines when you next rebase this PR: https://github.com/Homebrew/homebrew-bundle/blob/master/azure-pipelines.yml#L10-L12

**Comment by colindean** [comment]
I'm struggling to come up with the best way to create `.rubocop_todo.yml` as well as how to _use_ it if it was created.

## Creating the todo

Rubocop itself has `--auto-gen-config`. However, this seems to error when cops aren't available:

```
$ rubocop --auto-gen-config -c .rubocop_todo.yml
uninitialized constant RuboCop::Cop::RangeHelp
Did you mean?  Range
/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/ruby/2.3.0/gems/rubocop-rspec-1.30.0/lib/rubocop/rspec/blank_line_separation.rb:7:in `<module:BlankLineSeparation>'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/ruby/2.3.0/gems/rubocop-rspec-1.30.0/lib/rubocop/rspec/blank_line_separation.rb:5:in `<module:RSpec>'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/ruby/2.3.0/gems/rubocop-rspec-1.30.0/lib/rubocop/rspec/blank_line_separation.rb:2:in `<module:RuboCop>'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/ruby/2.3.0/gems/rubocop-rspec-1.30.0/lib/rubocop/rspec/blank_line_separation.rb:1:in `<top (required)>'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/ruby/2.3.0/gems/rubocop-rspec-1.30.0/lib/rubocop-rspec.rb:22:in `require_relative'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/ruby/2.3.0/gems/rubocop-rspec-1.30.0/lib/rubocop-rspec.rb:22:in `<top (required)>'
/Users/colin/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:120:in `require'
/Users/colin/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:120:in `require'
/usr/local/Homebrew/Library/Homebrew/rubocops.rb:3:in `<top (required)>'
<snip>
```
when I've manually created the todo and pointed it at the project-level config:
```
$ cat .rubocop_todo.yml
inherit_from:
- /usr/local/Homebrew/Library/.rubocop.yml
```

I tried hacking around inside [brew's `Library/Homebrew/style.rb`](https://github.com/Homebrew/brew/blob/a4c62d37b1a143109f631a062c3dba21f462d9d4/Library/Homebrew/style.rb#L79-L84) to add an `--auto-gen-config` in there but it seemed to be _very_ unhappy when I did that and basically easily-revertably hosed my installation of brew and bundle.

## Using the todo

I also don't see how a todo in the bundle project directory would be picked up.

https://github.com/Homebrew/brew/blob/a4c62d37b1a143109f631a062c3dba21f462d9d4/Library/Homebrew/style.rb#L65

indicates that really only `/usr/local/Homebrew/Library/.rubocop_audit.yml` is going to be set.

Is this something that needs to be added to brew before proceeding?

**Comment by colindean** [comment]
Another pass at the rubocop todo. I'm not sure I've executed this the way you want, @MikeMcQuaid. The path to Homebrew's rubocops seemed to need to be hardcoded…

**Comment by colindean** [comment]
Blah, looks like one of the autofixes messed up

**Comment by colindean** [comment]
I started on some of the easier cops in `refactor-spec-for-rubocop` on my fork.

**Comment by colindean** [comment]
This is pretty time-consuming. We're going to be chipping away at this for a while 😞

**Comment by colindean** [comment]
Putting this here for safekeeping: https://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/NamedSubject

**Comment by MikeMcQuaid** [comment]
@colindean Thanks for your work here ❤️. Could consider merging what you have here and opening an issue to track the rest?

**Comment by colindean** [comment]
Yes. Merged in my fork's PR and created #388 for tracking future work on the TODO.

**Comment by MikeMcQuaid** [comment]
Thanks @colindean!

File: azure-pipelines.yml [hunk 1]
```diff
@@ -10,6 +10,8 @@ jobs:
         # TODO: RuboCop needs fixed in
         # https://github.com/Homebrew/homebrew-bundle/pull/384
         # brew style homebrew/bundle
+        # Use local config until then
+        rubocop --force-exclusion --parallel --except FormulaAudit,FormulaAuditStrict,NewFormulaAudit --format simple
```
Comment (MikeMcQuaid) [inline azure-pipelines.yml]: @colindean This seems like it's maybe not working as expected: https://dev.azure.com/Homebrew/Homebrew/_build/results?buildId=2783

It's outputting failures but not failing the build.

Comment (colindean) [reply azure-pipelines.yml]: Running `rubocop --force-exclusion --parallel --except FormulaAudit,FormulaAuditStrict,NewFormulaAudit --format simple` locally exits 1. 

I wonder if this bash block needs a `set -e` at the beginning? I'll give it a shot.

Comment (colindean) [reply azure-pipelines.yml]: When I run `brew style homebrew/bundle`, those alignment warnings aren't present. Maybe `.rubocop.yml` in this repo and the main repo have diverged further. If I run autofix with the local `.rubocop.yml`, the code style will then be in conflict with upstream. I don't think we want that. If we add `set -e`, then the current state will fail and we won't be able to merge anything until we've accepted that conflict or completed #388…

Comment (colindean) [reply azure-pipelines.yml]: https://github.com/Homebrew/homebrew-bundle/pull/436 makes it exit but I'm not sure we should merge that yet.

Comment (colindean) [reply azure-pipelines.yml]: See #437.

Comment (MikeMcQuaid) [reply azure-pipelines.yml]: Thanks ❤️

**Pull request merged**